app/src/main/java/com/edulectronics/tinycircuit/Views/CircuitActivity.java

```diff
@@ -189,8 +193,13 @@ public void onClick(View v) {
         if (!isInWireMode) {
             // Let clicked component handle the tap.
             if (circuitController.handleClick(((GridCell) v).mCellNumber)) {
-                ((GridCell) v).resetImage();
-                run(null);
+                Component component = circuitController.getComponent(((GridCell) v).mCellNumber);
```

Comment:
mooier zou zijn om deze componenten een IClickable interface te laten implementeren, die een handleClick() method heeft, die dit afhandelt. Dan hoef je hiier niet uit te zoeken wat voor component het is en verplaats je de code naar het component zelf.